algebird-core/src/main/scala/com/twitter/algebird/experimental/BloomFilter.scala

```diff
@@ -0,0 +1,402 @@
+      } else {
+        (this + other, ApproximateBoolean.exactFalse)
+      }
+
+    override def contains(x: A): ApproximateBoolean = ApproximateBoolean.exact(item == x)
+
+    override def maybeContains(x: A): Boolean = item == x
+
+    override def size: Approximate[Long] = Approximate.exact[Long](1)
+  }
+
+  /*
+   * Bloom filter with multiple values
+   */
+  case class Instance(bits: BitSet) extends BF {
+
+    /**
+     * The number of bits set to true
+     */
+    override def numBits: Int = bits.size.toInt
+
+    override val toBitSet: BitSet = bits
+
+    override def ++(other: BF): BF =
+      other match {
+        case _: Zero.type        => this
+        case Item(item)          => this + item
+        case Instance(otherBits) => Instance(bits | otherBits)
+      }
+
+    override def +(item: A): Instance = Instance(bits | BitSet(Hash(item)))
+
+    override def checkAndAdd(other: A): (BF, ApproximateBoolean) =
+      (this + other, contains(other))
+
+    override def maybeContains(item: A): Boolean = {
+      val il = Hash(item)
+      var idx = 0
+      var found = true
+      while (idx < il.length && found) {
+        val i = il(idx)
+        if (!bits(i)) {
+          found = false
+        }
+        idx += 1
+      }
+      found
+    }
+
+    // use an approximation width of 0.05
+    override def size: Approximate[Long] =
+      BloomFilter.sizeEstimate(numBits, numHashes, width, 0.05)
+  }
+
+  implicit val monoid = new Monoid[BF] with BoundedSemilattice[BF] {
+    override val zero: BF = Zero
+
+    /**
+     * Assume the bloom filters are compatible (same width and same hashing functions).  This
+     * is the union of the 2 bloom filters.
+     */
+    override def plus(left: BF, right: BF): BF = left ++ right
+
+    override def sumOption(as: TraversableOnce[BF]): Option[BF] =
+      if (as.iterator.isEmpty) None
+      else {
+        var bfItem: Item = null
+        val iter = as.iterator
+        var counter = 0
+        var bs = BitSet.newEmpty(0)
+
+        while (iter.hasNext) {
+          iter.next() match {
+            case _: Zero.type => ()
+            case bi @ Item(item) =>
+              bfItem = bi
+              val array = Hash(item)
+              counter += array.length
+              bs = bs.mutableAdd(array)
+            case Instance(bitset) =>
+              val iter = bitset.iterator
+              while (iter.hasNext) {
+                counter += 1
+                bs = bs.mutableAdd(iter.next())
+              }
+          }
+        }
+
+        if (counter == 0) Some(zero)
+        else if (counter == n && (bfItem != null)) Some(bfItem)
+        else Some(Instance(bs))
+      }
+  }
+
+  implicit val aggregator = new MonoidAggregator[A, BF, BF] {
+    override val monoid: Monoid[BF] = self.monoid
+
+    override def prepare(value: A): BF = Item(value)
+
+    override def present(bf: BF): BF = bf
+  }
+
+  implicit val equiv = new Equiv[BF] {
```

Comment:
I'll add it, in this cases I tend to be ok with not having since it's type is clear... We should probably add this rule to scalafix to check for explicit result types.